### PR TITLE
Fix an error related to empty ensure bodies

### PR DESCRIPTION
**Pull request by presidentbeef**
Addresses #306

### CODE REVIEW

File: lib/ruby_parser_extras.rb [hunk 1]
```diff
@@ -723,7 +723,9 @@ def new_body val
       result = block_append(result, elsebody)
     end
 
-    result = s(:ensure, result, ensurebody).compact.line result.line if ensurebody
+    if ensurebody
+      result = s(:ensure, result, ensurebody).compact.line (result || ensurebody).line
```
Comment (zenspider) [inline lib/ruby_parser_extras.rb:727]: I split this out to a local variable because I found the "space open-paren" part confusing.

File: test/test_ruby_parser.rb [hunk 2]
```diff
@@ -185,6 +185,13 @@ def test_begin_else_return_value
     assert_syntax_error rb, "else without rescue is useless"
   end
 
+  def test_begin_rescue_ensure_no_bodies
+    rb = "begin\nensure\nend"
+    pt = s(:ensure, s(:nil).line(2)).line(2)
+
+    assert_parse rb, pt
+  end
```
Comment (zenspider) [inline test/test_ruby_parser.rb:193]: and added a bunch more tests w/ more coverage across all these crazy edge cases.

**Comment by zenspider** [comment]
Added w/ some changes... gonna make one more (empty resbody `nil` -> `s(:nil)`)

File: lib/ruby_parser_extras.rb (hunk 1 above)
Comment (presidentbeef) [reply lib/ruby_parser_extras.rb:727]: Absolutely :+1:
